File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.py
def solution(A, B):
    answer: int = 0
    A.sort()
    B.sort()
    i: int = 0
    j: int = 0
    
    while j < len(B):
        if A[i] < B[j]:
            i += 1
            answer += 1
        j += 1
        
    return answer

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.java
import java.util.*;

class Solution {
    public int solution(int[] A, int[] B) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int num : B) pq.offer(num);
        Arrays.sort(A);
        int idx = 0;
        int n = A.length;
        int answer = 0;
        while (idx < n && !pq.isEmpty()) {
            int aNum = A[idx];
            int bNum = pq.poll();
            if (aNum < bNum) {
                answer++;
                idx++;
            }
        }
        return answer;
    }
}